upb/upb.h

```diff
@@ -196,10 +196,17 @@ template <int N> class InlinedEnvironment;
 
 #define UPB_UNUSED(var) (void)var
 
-/* For asserting something about a variable when the variable is not used for
- * anything else.  This prevents "unused variable" warnings when compiling in
- * debug mode. */
-#define UPB_ASSERT_VAR(var, predicate) UPB_UNUSED(var); assert(predicate)
+/* UPB_ASSERT(): in release mode, we use the expression without letting it be
+ * evaluated.  This prevents "unused variable" warnings. */
+#ifdef NDEBUG
+#define UPB_ASSERT(expr) if (false && (expr)) {}
```

Comment:
`(void) var` evaluates the expression and throws away the result. It's important that `UPB_ASSERT()` in release mode *not* evaluate the expression. When I change this to use `UPB_UNUSED()` instead, it grows the binary in release mode (and presumably slows it down too).